server/src/server.ts

```diff
@@ -74,6 +78,127 @@ connection.onInitialize(
     },
 );
 
+connection.onInitialized((_params) => {
+    connection.workspace.getWorkspaceFolders().then((_value) => {
+        const workspacePath = url.fileURLToPath(_value[0].uri);
+        glob('**/project.kao', {
+            cwd: workspacePath,
+        })
+            .then((_matches) => {
+                for (const filePath of _matches) {
+                    if (openProjectFile(workspacePath, filePath)) {
+                        // There should be only one `project.kao` file. If we found a good candidate, stop the loop.
+                        return;
+                    }
+                }
+            })
+            .catch((_error) => console.error(_error));
+        /*
+         * List of the file-extensions of yseopml files that are never set by the user in `project.kao`-like files.
+         * This list is a subset of the file-extensions known by this extension as set in `client/package.json`.
+         */
+        const yseopmlExtensions = ['yclass', 'yobject', 'ycomplete'];
+        for (const extension of yseopmlExtensions) {
+            parseFilesWithExtension(workspacePath, extension);
+        }
+    });
+});
+
+/**
+ * Find all the files in the workspace that have the extension `extension`
+ * and open them as `TextDocument` objects. This will result in a parsing request for
+ * these files and have it known by the extension.
+ * This function excludes the results from `.generated-yml/`.
+ *
+ * @param extension The extension of the files to look for
+ */
+function parseFilesWithExtension(workspacePath: string, extension: string): void {
+    glob(`**/*.${extension}`, {
+        cwd: workspacePath,
+        ignore: '**/.generated-yml/**',
+    })
+        .then((_matches) => {
+            if (!_matches) {
+                return;
+            }
+            _matches.forEach((uri) => {
+                fsPromises
+                    .readFile(uri)
+                    .then((_doc) => {
+                        const doc = _doc.toString();
+                        parseFile(`file://${workspacePath}/${uri}`, doc);
+                    })
+                    .catch((_err) => {
+                        if (!!_err) {
+                            connection.console.error(`${_err}`);
+                            return;
```

Comment:
I see the `return` is in the `then`. Is it intended, like “abort for all files as soon as one file makes things go wrong”?
Btw, how likely are we to `catch` an error that would be falsey? And is it better to ignore falsey errors (provided that such things exist) than to log them anyway? This `if` seems strange.

Gosh, sometimes those reviews sound more like you’re teaching me TS than proper reviews.

Reply:
> I see the return is in the then. Is it intended, like “abort for all files as soon as one file makes things go wrong”?

The `return` is in the `catch`, but I will remove it, it is useless. This is what remains of a former implementation.

> Btw, how likely are we to catch an error that would be falsey? And is it better to ignore falsey errors (provided that such things exist) than to log them anyway? This if seems strange.

I don't think we can get a falsy error here, but since we can get `any` here I have to do the check, I think.
I can still log the fact that an unknown error occurred, what do you think?

> Gosh, sometimes those reviews sound more like you’re teaching me TS than proper reviews.

Well, I'm a YML dev, not a TS dev. I believe we are both teaching to each other. :)

Reply:
Yeah we could at least add in the `console.error` call a couple of words about the context (`File parsing: …` or whatever; I don’t recall the exact context we had here) and make sure we don’t use the auto `${…}` toString-ish thing on an empty string, a zero, a NaN… xD